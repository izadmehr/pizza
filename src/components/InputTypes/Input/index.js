import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Label from "../Label";
import ErrorText from "../ErrorText";

export const StyledInput = styled.input`
  padding: 10px 10px;
  border: 1px solid ${props => props.theme.color.lightGrey};
  margin-top: 10px;
`;

const Input = ({
  id,
  label,
  values,
  touched,
  errors,
  type,
  placeholder,
  handleChange,
  handleBlur
}) => (
  <Label htmlFor={id}>
    {label}
    <StyledInput
      id={id}
      name={id}
      value={values[id]}
      type={type}
      placeholder={placeholder}
      onChange={handleChange}
      onBlur={handleBlur}
    />
    {errors[id] && touched[id] && <ErrorText>{errors[id]}</ErrorText>}
  </Label>
);

Input.defaultProps = {
  type: "text",
  placeholder: "",
  label: "",
  id: ""
};

Input.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  values: PropTypes.object.isRequired,
  touched: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired
};

export default Input;
