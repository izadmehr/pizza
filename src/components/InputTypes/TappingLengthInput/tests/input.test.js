import React from "react";
import { shallow } from "enzyme";

import Input from "../index";

describe("Input Component", () => {
  const fakeProps = {
    id: "fakeId",
    type: "text",
    values: {
      fakeId: "fakeValue"
    },
    placeholder: "fake placeholder",
    handleChange: () => ({}),
    handleBlur: () => ({})
  };

  it("renders with all props", () => {
    const wrapper = shallow(
      <Input
        touched={{
          fakeId: true
        }}
        errors={{
          fakeId: "fake error message"
        }}
        {...fakeProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without error message", () => {
    const wrapper = shallow(<Input errors={{}} touched={{}} {...fakeProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
