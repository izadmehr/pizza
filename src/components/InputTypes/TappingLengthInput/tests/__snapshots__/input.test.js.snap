// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders with all props 1`] = `
<Fragment>
  <styled.label
    htmlFor="fakeId"
  >
    <styled.input
      id="fakeId"
      name="fakeId"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="fake placeholder"
      value="1"
    />
  </styled.label>
</Fragment>
`;

exports[`Input Component renders without error message 1`] = `
<Fragment>
  <styled.label
    htmlFor="fakeId"
  >
    <styled.input
      id="fakeId"
      name="fakeId"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="fake placeholder"
      value="1"
    />
  </styled.label>
</Fragment>
`;
